### DISCUSSION TITLE
Are tuples supported?

### DISCUSSION THREAD

**Discussion by digitalnovats** [vazco/uniforms#1336] [Q&A]
I would like to know if tuples are supported. at the moment, it seems to me that there is no such support. In fact in the following case: 

```
const schema = {
    title: "Address",
    type: "object",
    properties: {
        city: { type: "string" },
        state: { type: "string" },
        street: { type: "string" },
        zip: { type: "string", pattern: "[0-9]{5}" },
        arr: {
            type: "array",
            items: [{ type: "string" }],
        },
    },
    required: ["street", "zip", "state"],
};
```

It gives me the error "Invariant Violation: Field not found in schema: "arr.$"". 

Does anyone have any idea how to use a scheme like the one above??

**Comment by kestarumper** [comment]
Hi @digitalnovats,
Currently, tuples are not supported, or it might be a bug (not sure). However, you could use a workaround where you specify `items` as `{ type: "object" }` and add properties.

I also noticed that if you do that, but use the `0`, `1` etc as property name, it will also throw an error.
[Error reproduction](https://uniforms.tools/playground/#?N4IgDgTgpgzlAuIBcICCATd0YwAQAoApAZQHkA5YgYwAsoBbAQwEoQAacCAezBmVEYBXeFxiMAblGQAzRgBs4HISLGSAIlDmMAnsgCMABgMd0ASzEAjOVHQz5ikFoubk8CIKgcwWqlBpc5dCgIOwVPEGhGdFIAOzldJFkwjhh_AHcASTjTGKgAUQhuEMT7cMYYbRiqWOJBC3pTRBLkkHLK6piANXlTdEZ4KWaHGFoGRmQQfHxmXABeAD5cYAAdGNxcKi4YmHhcRgArcTncXLTcVEP8YD25OQKimCRcNw82XEE4DVlBOXhH5_cnlwAGsoNo0lwIOh_gBtZYgQQxUzSSH0GDwgC6uAAvswANyrdabba7EZ0JjHFZrdbPRrWJ7wjBYWDo9iEmnwbRgKAMkBcCz7KBUeDwtjs9aQHjBeCmWBPKk0xUbRraeXPLk83DwnYQHIAc3hOLF1KVuB2_U11053N5Ov1huxxtN6x1UAQautmu1bntICN4qVAC9TGAPRrbT6Ygb2LgwP0BhAYryYQYALQATgxwAArNiHU7nYxCvKA871Tata1CjpRaXnY0GP8reHK_zBcLRbHuNyIDK5UtcAYwxXvbqo_ncHph16QDFBPRnBAHTj_SalY6Axu19AAI6CUzQdBPOEgV0ITvw4NgC-n-AWzHG7EEmLs6SI4WmLYbSIDbpyXr9JC-BkmMMwKjSxI7Lg4g9H0IgQMcBziAAdJs9BgKY1jAaMTD4qsAbQPAgiJgQ9BcEEcgzAsSx1jB_5wUBZEUXhL5rusyIEHRAHwchwQPLgABkAnQbBgEQLxhSQjAyHWFG8A0GBdbrIRxFrNcQR3ph_xcQx4l8VJOLPqaeZrk-7ImeykG7DpYnHFQP5QH-3FASBuHPuyKkkacuAkBQ1DkowABCuroHqUBXGaOGMG8NnwTiLG4tMICOiA8kMIMrQxPAtjYkAA).
